### ISSUE TITLE
Allow to pass environment variables into bundle configuration

### ISSUE THREAD

**Issue by edefimov** [overblog/GraphQLBundle#172]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Version/Branch   | 0.8.6

For now it is impossible to pass environment variables into bundle configuration due to variable type checks:
```
Uncaught Symfony\Component\Config\Definition\Exception\InvalidTypeException: Invalid type for path "overblog_graphql.security.query_max_depth". Expected int, but got string.
```
I think it would be useful to allow end-users to set up any scalar value in the bundle config using environment.

**Comment by mcg-web** [comment]
released in v0.8.7, thank you :+1:

**Comment by edefimov** [comment]
#173 Doesn't work with actual environment variables. When container is compiled, symfony replaces configuration value with the string like _env_MY_VARIABLE_hash_, which won't pass integer validator in configuration definition.

**Comment by mcg-web** [comment]
This should work with [symfony env](https://symfony.com/doc/current/configuration/external_parameters.html), that's what you talking of ?

**Comment by edefimov** [comment]
@mcg-web Yes, that is

**Comment by mcg-web** [comment]
on compilation `%en(MY_VARIABLE)%` is replaced by `$this->getEnv('MY_VARIABLE')` and not by string, I don't understand why validation will fail... Can you explain me why please? I don't get it sorry

**Comment by edefimov** [comment]
During the compilation stage environment variables, defined in config file, are converted into placeholders like _env_MY_VARIABLE_hash_  and this placeholder is passed into normalization and validation functions of config definition. So when configuration node is defined like `integerNode`, `booleanNode` or any other node, except `scalarNode` one, type validation will either fail or cast placeholder into corresponding node type. Silent casting leads to bugs when services do not receive actual values.
When container is dumped into php files placeholders receives actual values, using call `$this->getEnv('MY_VARIABLE')`.

So the correct solution should look like:
1. Define configuration node as `scalarNode`
2. Keep normalizers the same as in #173  
3. Perform actual type casting in target services, which use these config variables in arguments
4. Add test when configuration value is defined like environment variable

**Comment by mcg-web** [comment]
thank you @edefimov, can you submit a PR for this please?

**Comment by edefimov** [comment]
Sure: #176

**Comment by mcg-web** [comment]
thanks again for the contribution, released in v0.8.8...
